pinning-services/index.js

```diff
@@ -81,73 +93,130 @@ const job = new cron.CronJob(period, function() {
-        if (fs.existsSync(dirApi)) {
-          getDatSeed('dat-seed-api', projName, domain, txtHypercoreApi)
-            .then(seed => hyperPublisher.sync({
-              seed,
-              fsPath: dirApi,
-              drivePath: '/'
-            }))
-            .then(({diff, url}) => {
-              // Log any changes to hyperdrive and return url
-              console.log(`API responses for ${projName} pinned at ${url}. Changes:`);
-              console.log(diff);
-              return url;
-            })
-            .catch(function(error) {
-              console.log(error);
-            });
+        if (protoHypercore) {
+          if (fs.existsSync(dirApi)) {
+            getDatSeed('dat-seed-api', projName, domain, txtHypercoreApi)
+              .then(seed => hyperPublisher.sync({
+                seed,
+                fsPath: dirApi,
+                drivePath: '/'
+              }))
+              .then(({ diff, url }) => {
+                // Log any changes to hyperdrive and return url
+                console.log(`API responses for ${projName} pinned at ${url}. Changes:`);
+                console.log(diff);
+                return url;
+              })
+              .catch(function(error) {
+                console.log(error);
+              });
+          }
         }
 
         // Pin API responses to IPFS
-        if (fs.existsSync(dirApi)) {
-          ipfs.add(globSource(dirApi, { recursive: true, followSymlinks: false }), { cidVersion: 1, pin: true, timeout: 60000 })
-            .then(file => file['cid'])
-            .then(cid => {
-              console.log(`API responses for ${projName} pinned at ipfs/${cid}`);
-
-              // Update DNS record
-              return updateDnsRecordDigitalOcean(domain, 'TXT', txtIpfsApi, `dnslink=/ipfs/${cid}`, 300, conf['digitalOceanAccessToken']);
-            })
-            .catch(function(error) {
-              console.log(error);
-            });
+        if (protoIpfs) {
+          if (fs.existsSync(dirApi)) {
+            ipfs.add(globSource(dirApi, { recursive: true, followSymlinks: false }), { cidVersion: 1, pin: true, timeout: 60000 })
+              .then(file => file['cid'])
+              .then(cid => {
+                console.log(`API responses for ${projName} pinned at ipfs/${cid}`);
+
+                // Update DNS record
+                return updateDnsRecordDigitalOcean(domain, 'TXT', txtIpfsApi, `dnslink=/ipfs/${cid}`, 300, conf['digitalOceanAccessToken']);
+              })
+              .catch(function(error) {
+                console.log(error);
+              });
+          }
+        } else {
+          if (protoIpfsPurgeIfDisabled)
+            updateDnsRecordDigitalOcean(domain, 'TXT', txtIpfsApi, '', 300, conf['digitalOceanAccessToken']);
+        }
+
+        // Update HTTP A and AAAA record
+        if (protoHttp) {
+          if (httpPublisherIpv6Address && httpPublisherIpv6Address.trim().length)
```

Comment:
I'd trim at the time of assignment, rather than time of use, this way we aren't floating around possibly dirty vars.